packages/application-components/src/components/modal-page/modal-page.js

```diff
@@ -29,136 +28,74 @@ const parentSelector = () =>
-            <Spacings.Inline
-              scale="m"
-              alignItems="center"
-              justifyContent="space-between"
-            >
-              <Text.Headline
-                elementType="h3"
-                truncate={true}
-                title={props.title}
-              >
-                {props.title}
-              </Text.Headline>
-              {props.onClose && (
-                <SecondaryIconButton
-                  label="Close page"
-                  onClick={props.onClose}
-                  icon={<CloseIcon size="medium" />}
-                />
-              )}
-            </Spacings.Inline>
-            <Spacings.Inline
-              scale="m"
-              alignItems="center"
-              justifyContent="space-between"
-            >
-              {props.subtitle}
-              <div>{props.components.actions}</div>
-            </Spacings.Inline>
-          </div>
-          <div className={styles['page-content']}>{props.children}</div>
-        </Spacings.Stack>
-      </Card>
-    </div>
-  </Modal>
+  <ClassNames>
+    {({ css: makeClassName }) => (
+      <Modal
+        isOpen={props.isOpen}
+        className={makeClassName(getContainerStyles(props))}
+        ariaHideApp={false}
+        contentLabel={props.title}
+        onRequestClose={props.onClose}
+        parentSelector={props.parentSelector || parentSelector}
+        overlayClassName={makeClassName(getOverlayStyles(props))}
+        shouldCloseOnEsc={Boolean(props.onClose)}
+        shouldCloseOnOverlayClick={Boolean(props.onClose)}
+      >
+        {props.showTopBar && (
+          <ModalPageTopBar
+            onClose={props.onClose}
+            currentPath={props.topBarLabels.currentPath || props.title}
+            previousPath={props.topBarLabels.previousPath}
+            tone={props.topBarTone}
+          />
+        )}
+
+        {props.showHeader && (
+          <ModalPageHeader
+            title={props.title}
+            actions={props.headerActions}
+            subtitle={props.subtitle}
+          />
+        )}
+        <div
+          css={css`
+            overflow: auto;
+            flex: 1;
+          `}
+        >
+          {props.children}
+        </div>
+      </Modal>
+    )}
+  </ClassNames>
 );
 ModalPage.displayName = 'ModalPage';
 ModalPage.propTypes = {
+  level: PropTypes.number,
+  title: PropTypes.string.isRequired,
+  zIndex: PropTypes.number,
   isOpen: PropTypes.bool.isRequired,
   onClose: PropTypes.func,
-  zIndex: PropTypes.number,
+  subtitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  children: PropTypes.node.isRequired,
+  showTopBar: PropTypes.bool,
+  showHeader: PropTypes.bool,
   baseZIndex: PropTypes.number,
-  title: PropTypes.string.isRequired,
-  subtitle: PropTypes.node,
-  components: PropTypes.shape({
-    actions: PropTypes.node,
+  topBarTone: PropTypes.oneOf(['white', 'gray']),
+  topBarLabels: PropTypes.shape({
```

Comment:
Nit: we often prefer flat props to avoid people passing in new object references. so `topBarCurrentPathLabel` etc. The sum of character needing typing is the same after all.

Reply:
Yeah, this was one of the things I wasn't so sure myself and was awaiting feedback.